### PR TITLE
feat: Add hidden properties list processEvent plugins can't see

### PR DESCRIPTION
**Pull request by tiina303**
## Problem

As new products are getting built on top of events and things are added to the ingestion pipeline we don't want plugins to get more data nor users to start to depend on stuff.

Related: 
- https://github.com/PostHog/flatten-properties-plugin/pull/8
- https://posthog.slack.com/archives/C0374DA782U/p1714568052844489

## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: plugin-server/src/worker/ingestion/event-pipeline/pluginsProcessEventStep.ts [hunk 1]
```diff
@@ -5,22 +5,48 @@ import { runProcessEvent } from '../../plugins/run'
 import { droppedEventCounter } from './metrics'
 import { EventPipelineRunner } from './runner'
 
+// Some properties are internal, which plugins shouldn't be able to change
+// We don't want these to be passed to the processEvent plugins
+const hiddenProperties = [
+    '$heatmap_data', // Is processed and removed at a later pipeline step
+    '$elements', // This is deprecated
+    '$groups', // To avoid dependencies if we want to change how we use this property
+    '$active_feature_flags', // To avoid dependencies if we want to change how we use this property
```
Comment (bretthoerner) [inline plugin-server/src/worker/ingestion/event-pipeline/pluginsProcessEventStep.ts:14]: I take it that you just know these aren't used in any active plugins?

**Review by bretthoerner — APPROVED** [review]

**Comment by bretthoerner** [comment]
I think there may be real test failures.

File: plugin-server/src/worker/ingestion/event-pipeline/pluginsProcessEventStep.ts (hunk 1 above)
Comment (tiina303) [reply plugin-server/src/worker/ingestion/event-pipeline/pluginsProcessEventStep.ts:14]: Better safe than sorry (good call out), so double checking:
Code:
- `$heatmap_data` - is new so good
- `$elements` - is mentioned in code by 6 plugins 4 destinations [metabase](http://metabase-prod-us/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7InR5cGUiOiJuYXRpdmUiLCJuYXRpdmUiOnsicXVlcnkiOiJzZWxlY3QgKiBmcm9tIHBvc3Rob2dfcGx1Z2luIHdoZXJlIGlkIGluIChcbnNlbGVjdCBwbHVnaW5faWQgZnJvbSBwb3N0aG9nX3BsdWdpbnNvdXJjZWZpbGUgd2hlcmUgc291cmNlIExJS0UgJyUkZWxlbWVudHMlJ1xuKSIsInRlbXBsYXRlLXRhZ3MiOnt9fSwiZGF0YWJhc2UiOjM0fSwiZGlzcGxheSI6InRhYmxlIiwicGFyYW1ldGVycyI6W10sInZpc3VhbGl6YXRpb25fc2V0dGluZ3MiOnt9fQ==), which get no changes from this, property_flatener that is already handled by linked PR and netdata, which I thought was done, but I'm wrong https://github.com/PostHog/posthog/issues/22020 so I'll remove elements from here
- $groups - only property flattener processEvent (denylisting there explicitly) 
- $active_feature_flags - only property flattener processEvent (denylisting there explicitly) 

Secondly folks could be using this within their configs:
- pluginConfigs via equivalent of `select * from posthog_pluginconfig where config::text like '%$groups%'`: there's a single user in EU setting `$active_feature_flags` to null via propertyFilter, they aren't exporting it though, so only within posthog, asked about this in https://posthog.slack.com/archives/C034XD440RK/p1714584381832039

Thirdly folks could be using it within plugin config attachments, don't know how to check these effectively, so I'm going to just go with `$heatmap_data` only for now to be safe.

**Comment by tiina303** [comment]
I'm gonna abandon this instead - it's adding some cpu work on top and this specific heatmap is handled in property flattener and hope that people won't do silly things here.

**Pull request closed without merging**